### ISSUE TITLE
[native-image] Invalid or corrupt jarfile truffle-nfi-native-linux-amd64-20.1.0.tar.gz

### ISSUE THREAD

**Issue by aviflax** [oracle/graal#2481]
**Describe the issue**

When running native-image on a Mac, and including `org.graalvm.nativeimage/svm` in the classpath for my project, compilation fails with this error:

```text
Error: Invalid or corrupt jarfile /Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz
Caused by: java.util.zip.ZipException: zip END header not found
com.oracle.svm.driver.NativeImage$NativeImageError: Invalid or corrupt jarfile /Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz
	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1537)
	at com.oracle.svm.driver.NativeImage.processJarManifestMainAttributes(NativeImage.java:986)
	at com.oracle.svm.driver.NativeImage.processManifestMainAttributes(NativeImage.java:973)
	at com.oracle.svm.driver.NativeImage.addImageClasspathEntry(NativeImage.java:1446)
…
```

**Steps to reproduce the issue**

I hope you don’t mind, I’ve reproduced the error in [this repo](https://github.com/FundingCircle/repro-graalvm-mac-issue/) and I’ve included a GitHub Actions job that runs on MacOS and  triggers the error consistently. For example [this run](https://github.com/FundingCircle/repro-graalvm-mac-issue/runs/690756479) is on a branch switching to 20.1.0 and it failed the same way. I’ve reproduced this now with 19.3.2 and 19.3.0.

**Describe GraalVM and your environment:**

 - CE 20.1.0 (and also 19.3.2)
 - JDK major version: 11
 - OS: MacOS 10.15
 - Architecture: AMD64

**More details**

The _full_ logs are available via [this Actions job run](https://github.com/FundingCircle/repro-graalvm-mac-issue/runs/690756479) but I’ve pasted what seems to be the most important bits here:

```
2020-05-19T22:04:46.3518550Z ##[group]Run bin/build-native-image
2020-05-19T22:04:46.3518800Z [36;1mbin/build-native-image[0m
2020-05-19T22:04:46.3673250Z shell: /bin/bash -e {0}
2020-05-19T22:04:46.3673430Z env:
2020-05-19T22:04:46.3673540Z   JAVA_HOME: /Users/runner/hostedtoolcache/GraalVM/20.1.0-java11/x64
2020-05-19T22:04:46.3673680Z   JAVA_HOME_20.1.0.java11: /Users/runner/hostedtoolcache/GraalVM/20.1.0-java11/x64
2020-05-19T22:04:46.3673820Z   CLOJURE_INSTALL_DIR: /Users/runner/hostedtoolcache/ClojureToolsDeps/1.10.1-536/x64/lib/clojure
2020-05-19T22:04:46.3673960Z ##[endgroup]
2020-05-19T22:04:46.4724680Z + rm -Rf classes
2020-05-19T22:04:46.4725030Z Clojure dependency tree:
2020-05-19T22:04:46.4731140Z + mkdir classes
2020-05-19T22:04:46.4779500Z + echo 'Clojure dependency tree:'
2020-05-19T22:04:46.4870040Z + clojure -Stree
2020-05-19T22:04:53.2032420Z Downloading: org/clojure/clojure/1.10.2-alpha1/clojure-1.10.2-alpha1.pom from central
2020-05-19T22:04:53.3921740Z Downloading: org/clojure/spec.alpha/0.2.187/spec.alpha-0.2.187.pom from central
2020-05-19T22:04:53.3957390Z Downloading: org/clojure/core.specs.alpha/0.2.44/core.specs.alpha-0.2.44.pom from central
2020-05-19T22:04:53.4469620Z Downloading: org/clojure/pom.contrib/0.3.0/pom.contrib-0.3.0.pom from central
2020-05-19T22:04:53.4536220Z Downloading: org/clojure/pom.contrib/0.2.2/pom.contrib-0.2.2.pom from central
2020-05-19T22:04:53.4653440Z Downloading: borkdude/clj-reflector-graal-java11-fix/0.0.1-graalvm-20.1.0/clj-reflector-graal-java11-fix-0.0.1-graalvm-20.1.0.pom from clojars
2020-05-19T22:04:53.6661510Z Downloading: org/graalvm/nativeimage/svm/20.1.0/svm-20.1.0.pom from central
2020-05-19T22:04:53.7483150Z Downloading: org/graalvm/nativeimage/svm-hosted-native-windows-amd64/20.1.0/svm-hosted-native-windows-amd64-20.1.0.pom from central
2020-05-19T22:04:53.7586610Z Downloading: org/graalvm/nativeimage/svm-hosted-native-linux-amd64/20.1.0/svm-hosted-native-linux-amd64-20.1.0.pom from central
2020-05-19T22:04:53.7690360Z Downloading: org/graalvm/nativeimage/svm-hosted-native-darwin-amd64/20.1.0/svm-hosted-native-darwin-amd64-20.1.0.pom from central
2020-05-19T22:04:53.7793430Z Downloading: org/graalvm/sdk/graal-sdk/20.1.0/graal-sdk-20.1.0.pom from central
2020-05-19T22:04:53.8136640Z Downloading: org/graalvm/truffle/truffle-nfi/20.1.0/truffle-nfi-20.1.0.pom from central
2020-05-19T22:04:53.8190790Z Downloading: org/graalvm/nativeimage/objectfile/20.1.0/objectfile-20.1.0.pom from central
2020-05-19T22:04:53.8294550Z Downloading: org/graalvm/nativeimage/pointsto/20.1.0/pointsto-20.1.0.pom from central
2020-05-19T22:04:53.8364150Z Downloading: org/graalvm/compiler/compiler/20.1.0/compiler-20.1.0.pom from central
2020-05-19T22:04:53.9117410Z Downloading: org/graalvm/truffle/truffle-api/20.1.0/truffle-api-20.1.0.pom from central
2020-05-19T22:04:53.9174040Z Downloading: org/graalvm/truffle/truffle-nfi-native-linux-aarch64/20.1.0/truffle-nfi-native-linux-aarch64-20.1.0.pom from central
2020-05-19T22:04:53.9183360Z Downloading: org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.pom from central
2020-05-19T22:04:53.9870310Z Downloading: org/graalvm/truffle/truffle-nfi-native-darwin-amd64/20.1.0/truffle-nfi-native-darwin-amd64-20.1.0.pom from central
2020-05-19T22:04:54.0477410Z Downloading: org/clojure/core.specs.alpha/0.2.44/core.specs.alpha-0.2.44.jar from central
2020-05-19T22:04:54.0555480Z Downloading: org/clojure/clojure/1.10.2-alpha1/clojure-1.10.2-alpha1.jar from central
2020-05-19T22:04:54.0663980Z Downloading: org/clojure/spec.alpha/0.2.187/spec.alpha-0.2.187.jar from central
2020-05-19T22:04:54.1012620Z Downloading: org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz from central
2020-05-19T22:04:54.1988100Z Downloading: org/graalvm/nativeimage/svm-hosted-native-linux-amd64/20.1.0/svm-hosted-native-linux-amd64-20.1.0.tar.gz from central
2020-05-19T22:04:54.2198080Z Downloading: borkdude/clj-reflector-graal-java11-fix/0.0.1-graalvm-20.1.0/clj-reflector-graal-java11-fix-0.0.1-graalvm-20.1.0.jar from clojars
2020-05-19T22:04:54.2952520Z Downloading: org/graalvm/truffle/truffle-nfi-native-linux-aarch64/20.1.0/truffle-nfi-native-linux-aarch64-20.1.0.tar.gz from central
2020-05-19T22:04:54.3353560Z Downloading: org/graalvm/nativeimage/pointsto/20.1.0/pointsto-20.1.0.jar from central
2020-05-19T22:04:54.4322600Z Downloading: org/graalvm/sdk/graal-sdk/20.1.0/graal-sdk-20.1.0.jar from central
2020-05-19T22:04:54.5707390Z Downloading: org/graalvm/truffle/truffle-nfi/20.1.0/truffle-nfi-20.1.0.jar from central
2020-05-19T22:04:54.5965290Z Downloading: org/graalvm/nativeimage/svm/20.1.0/svm-20.1.0.jar from central
2020-05-19T22:04:54.6046610Z Downloading: org/graalvm/nativeimage/svm-hosted-native-windows-amd64/20.1.0/svm-hosted-native-windows-amd64-20.1.0.tar.gz from central
2020-05-19T22:04:54.6798920Z Downloading: org/graalvm/nativeimage/svm-hosted-native-darwin-amd64/20.1.0/svm-hosted-native-darwin-amd64-20.1.0.tar.gz from central
2020-05-19T22:04:54.7550490Z Downloading: org/graalvm/nativeimage/objectfile/20.1.0/objectfile-20.1.0.jar from central
2020-05-19T22:04:54.8495710Z Downloading: org/graalvm/truffle/truffle-api/20.1.0/truffle-api-20.1.0.jar from central
2020-05-19T22:04:54.8729590Z Downloading: org/graalvm/truffle/truffle-nfi-native-darwin-amd64/20.1.0/truffle-nfi-native-darwin-amd64-20.1.0.tar.gz from central
2020-05-19T22:04:55.0740860Z Downloading: org/graalvm/compiler/compiler/20.1.0/compiler-20.1.0.jar from central
2020-05-19T22:05:00.3080680Z org.clojure/clojure 1.10.2-alpha1
2020-05-19T22:05:00.3128340Z   org.clojure/core.specs.alpha 0.2.44
2020-05-19T22:05:00.3129880Z   org.clojure/spec.alpha 0.2.187
2020-05-19T22:05:00.3131690Z borkdude/clj-reflector-graal-java11-fix 0.0.1-graalvm-20.1.0
2020-05-19T22:05:00.3133840Z   org.graalvm.nativeimage/svm 20.1.0
2020-05-19T22:05:00.3135130Z     org.graalvm.nativeimage/svm-hosted-native-linux-amd64 20.1.0
2020-05-19T22:05:00.3135630Z     org.graalvm.nativeimage/pointsto 20.1.0
2020-05-19T22:05:00.3137510Z     org.graalvm.sdk/graal-sdk 20.1.0
2020-05-19T22:05:00.3139270Z     org.graalvm.truffle/truffle-nfi 20.1.0
2020-05-19T22:05:00.3140100Z       org.graalvm.truffle/truffle-nfi-native-linux-amd64 20.1.0
2020-05-19T22:05:00.3151970Z       org.graalvm.truffle/truffle-nfi-native-linux-aarch64 20.1.0
2020-05-19T22:05:00.3153810Z       org.graalvm.truffle/truffle-api 20.1.0
2020-05-19T22:05:00.3181630Z       org.graalvm.truffle/truffle-nfi-native-darwin-amd64 20.1.0
2020-05-19T22:05:00.3182940Z     org.graalvm.nativeimage/svm-hosted-native-windows-amd64 20.1.0
2020-05-19T22:05:00.3184100Z     org.graalvm.nativeimage/svm-hosted-native-darwin-amd64 20.1.0
2020-05-19T22:05:00.3185710Z     org.graalvm.nativeimage/objectfile 20.1.0
2020-05-19T22:05:00.3190310Z     org.graalvm.compiler/compiler 20.1.0
2020-05-19T22:05:00.3426070Z + echo 'Compiling Clojure code to Java class files:'
2020-05-19T22:05:00.3426250Z Compiling Clojure code to Java class files:
2020-05-19T22:05:00.3429610Z + clojure -e '(compile '\''repro.core)'
2020-05-19T22:05:02.3670810Z repro.core
2020-05-19T22:05:02.3878290Z + echo 'Compiling class files to native image:'
2020-05-19T22:05:02.3878470Z Compiling class files to native image:
2020-05-19T22:05:02.3891790Z ++ clojure -Spath
2020-05-19T22:05:02.4109270Z + native-image -cp src:/Users/runner/.m2/repository/org/clojure/clojure/1.10.2-alpha1/clojure-1.10.2-alpha1.jar:/Users/runner/.m2/repository/borkdude/clj-reflector-graal-java11-fix/0.0.1-graalvm-20.1.0/clj-reflector-graal-java11-fix-0.0.1-graalvm-20.1.0.jar:/Users/runner/.m2/repository/org/clojure/core.specs.alpha/0.2.44/core.specs.alpha-0.2.44.jar:/Users/runner/.m2/repository/org/clojure/spec.alpha/0.2.187/spec.alpha-0.2.187.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-linux-amd64/20.1.0/svm-hosted-native-linux-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-aarch64/20.1.0/truffle-nfi-native-linux-aarch64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/pointsto/20.1.0/pointsto-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/sdk/graal-sdk/20.1.0/graal-sdk-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm/20.1.0/svm-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi/20.1.0/truffle-nfi-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-windows-amd64/20.1.0/svm-hosted-native-windows-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-darwin-amd64/20.1.0/svm-hosted-native-darwin-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/objectfile/20.1.0/objectfile-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-api/20.1.0/truffle-api-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-darwin-amd64/20.1.0/truffle-nfi-native-darwin-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/compiler/compiler/20.1.0/compiler-20.1.0.jar:classes -H:Name=repro -H:+ReportExceptionStackTraces --initialize-at-build-time --verbose --no-fallback --no-server -J-Xms6g -J-Xmx6g repro.core
2020-05-19T22:05:02.7384020Z Error: Invalid or corrupt jarfile /Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz
2020-05-19T22:05:02.7386190Z Caused by: java.util.zip.ZipException: zip END header not found
2020-05-19T22:05:02.7390380Z com.oracle.svm.driver.NativeImage$NativeImageError: Invalid or corrupt jarfile /Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz
2020-05-19T22:05:02.7391470Z 	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1537)
2020-05-19T22:05:02.7391730Z 	at com.oracle.svm.driver.NativeImage.processJarManifestMainAttributes(NativeImage.java:986)
2020-05-19T22:05:02.7392010Z 	at com.oracle.svm.driver.NativeImage.processManifestMainAttributes(NativeImage.java:973)
2020-05-19T22:05:02.7392250Z 	at com.oracle.svm.driver.NativeImage.addImageClasspathEntry(NativeImage.java:1446)
2020-05-19T22:05:02.7392490Z 	at com.oracle.svm.driver.NativeImage.addCustomImageClasspath(NativeImage.java:1416)
2020-05-19T22:05:02.7392680Z 	at com.oracle.svm.driver.DefaultOptionHandler.processClasspathArgs(DefaultOptionHandler.java:233)
2020-05-19T22:05:02.7393010Z 	at com.oracle.svm.driver.DefaultOptionHandler.consume(DefaultOptionHandler.java:104)
2020-05-19T22:05:02.7393280Z 	at com.oracle.svm.driver.NativeImage$NativeImageArgsProcessor.apply(NativeImage.java:1365)
2020-05-19T22:05:02.7393540Z 	at com.oracle.svm.driver.NativeImage.processNativeImageArgs(NativeImage.java:1581)
2020-05-19T22:05:02.7393770Z 	at com.oracle.svm.driver.NativeImage.completeImageBuild(NativeImage.java:1026)
2020-05-19T22:05:02.7393960Z 	at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1290)
2020-05-19T22:05:02.7394210Z 	at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1260)
2020-05-19T22:05:02.7394450Z 	at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1219)
2020-05-19T22:05:02.7394670Z 	at com.oracle.svm.driver.NativeImage$JDK9Plus.main(NativeImage.java:1722)
2020-05-19T22:05:02.7394850Z Caused by: java.util.zip.ZipException: zip END header not found
2020-05-19T22:05:02.7395070Z 	at java.util.zip.ZipFile$Source.zerror(ZipFile.java:1535)
2020-05-19T22:05:02.7395280Z 	at java.util.zip.ZipFile$Source.findEND(ZipFile.java:1436)
2020-05-19T22:05:02.7395510Z 	at java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1443)
2020-05-19T22:05:02.7396300Z 	at java.util.zip.ZipFile$Source.<init>(ZipFile.java:1274)
2020-05-19T22:05:02.7396500Z 	at java.util.zip.ZipFile$Source.get(ZipFile.java:1237)
2020-05-19T22:05:02.7396740Z 	at java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:727)
2020-05-19T22:05:02.7397130Z 	at java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:55)
2020-05-19T22:05:02.7398730Z 	at java.util.zip.ZipFile.<init>(ZipFile.java:247)
2020-05-19T22:05:02.7398870Z 	at java.util.zip.ZipFile.<init>(ZipFile.java:177)
2020-05-19T22:05:02.7399020Z 	at java.util.jar.JarFile.<init>(JarFile.java:348)
2020-05-19T22:05:02.7399300Z 	at java.util.jar.JarFile.<init>(JarFile.java:319)
2020-05-19T22:05:02.7399670Z 	at java.util.jar.JarFile.<init>(JarFile.java:285)
2020-05-19T22:05:02.7399810Z 	at com.oracle.svm.driver.NativeImage.processJarManifestMainAttributes(NativeImage.java:978)
2020-05-19T22:05:02.7399900Z 	... 12 more
2020-05-19T22:05:02.7403510Z ##[error]Process completed with exit code 1.
2020-05-19T22:05:02.7492510Z Post job cleanup.
```

**Comment by olpaw** [comment]
Why would you put a `.tar.gz` on your classpath?

**Comment by aviflax** [comment]
Well, *I* didn’t put it there; it comes from [the pom.xml](https://search.maven.org/artifact/org.graalvm.truffle/truffle-nfi/20.1.0/jar) for org.graalvm.truffle/truffle-nfi 20.1.0:

```xml
    <dependency>
      <groupId>org.graalvm.truffle</groupId>
      <artifactId>truffle-nfi-native-linux-amd64</artifactId>
      <version>20.1.0</version>
      <type>tar.gz</type>
    </dependency>
```

**Comment by olpaw** [comment]
Whatever produces ...
```
2020-05-19T22:05:02.3891790Z ++ clojure -Spath
[... THIS ...] ---> 2020-05-19T22:05:02.4109270Z + native-image -cp src:/Users/runner/.m2/repository/org/clojure/clojure/1.10.2-alpha1/clojure-1.10.2-alpha1.jar:/Users/runner/.m2/repository/borkdude/clj-reflector-graal-java11-fix/0.0.1-graalvm-20.1.0/clj-reflector-graal-java11-fix-0.0.1-graalvm-20.1.0.jar:/Users/runner/.m2/repository/org/clojure/core.specs.alpha/0.2.44/core.specs.alpha-0.2.44.jar:/Users/runner/.m2/repository/org/clojure/spec.alpha/0.2.187/spec.alpha-0.2.187.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-linux-amd64/20.1.0/svm-hosted-native-linux-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-aarch64/20.1.0/truffle-nfi-native-linux-aarch64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/pointsto/20.1.0/pointsto-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/sdk/graal-sdk/20.1.0/graal-sdk-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm/20.1.0/svm-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi/20.1.0/truffle-nfi-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-windows-amd64/20.1.0/svm-hosted-native-windows-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/svm-hosted-native-darwin-amd64/20.1.0/svm-hosted-native-darwin-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/nativeimage/objectfile/20.1.0/objectfile-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-api/20.1.0/truffle-api-20.1.0.jar:/Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-darwin-amd64/20.1.0/truffle-nfi-native-darwin-amd64-20.1.0.tar.gz:/Users/runner/.m2/repository/org/graalvm/compiler/compiler/20.1.0/compiler-20.1.0.jar:classes -H:Name=repro -H:+ReportExceptionStackTraces --initialize-at-build-time --verbose --no-fallback --no-server -J-Xms6g -J-Xmx6g repro.core
2020-05-19T22:05:02.7384020Z Error: Invalid or corrupt jarfile /Users/runner/.m2/repository/org/graalvm/truffle/truffle-nfi-native-linux-amd64/20.1.0/truffle-nfi-native-linux-amd64-20.1.0.tar.gz
```
... line needs to be fixed. It makes no sense for `native-image` to accept anything other than directories or jar-files as classpath entries.

**Comment by aviflax** [comment]
That makes sense. That’s the Clojure library [tools.deps.alpha](https://github.com/clojure/tools.deps.alpha/) — I’ll ask the maintainers about this. Thank you!
